### DISCUSSION TITLE
Use as esm without cdn

### DISCUSSION THREAD

**Discussion by MentalGear** [fingerprintjs/fingerprintjs#800] [Q&A] [answered]
Is it possible to use the library only on device without "phoning" to the cdn?

I just want to get a fingerprint to use, and it won't matter if accuracy drops.

**Comment by makma** [comment, marked as answer]
Hello @MentalGear, you can use the Webpack/Rollup/NPM/Yarn installation method documented [here](https://github.com/fingerprintjs/fingerprintjs/blob/master/docs/api.md#webpackrollupnpmyarn).

```sh
# Install the package first:
npm i @fingerprintjs/fingerprintjs
# or
yarn add @fingerprintjs/fingerprintjs
```

```javascript
import FingerprintJS from '@fingerprintjs/fingerprintjs'

// Initialize an agent at application startup.
const fpPromise = FingerprintJS.load()

;(async () => {
  // Get the visitor identifier when you need it.
  const fp = await fpPromise
  const result = await fp.get()
  console.log(result.visitorId)
})()
```

**Reply by MentalGear** [reply]
Thanks for the reply! So this won't trigger any telemetry to fingerprint.com and execute only locally?

**Reply by makma** [reply]
Yes, if you set it up accordingly (see below), it won't send any requests.
```javascript
const fpPromise = FingerprintJS.load({
  monitoring: false
})
```

**Reply by MentalGear** [reply]
Great ! Is there any indication available how much accuracy gets dropped with this method? I read on the main page, that the fingerprint method uses a mix of indicators, including users' IP address, which obviously wouldn't be available when running client-side only.

**Reply by makma** [reply]
> Is there any indication available how much accuracy gets dropped with this method?
 
It's hard to quantify accuracy drop precisely. Without the updated library, once there's a new change introduced in specific browser that affects fingerprinting logic or entropy sources, it might affect accuracy. Needless to say, browsers are evolving quite fast these days. To take a picture you can check our [release frequency](https://github.com/fingerprintjs/fingerprintjs/releases).

>  I read on the main page, that the fingerprint method uses a mix of indicators, including users' IP address, which obviously wouldn't be available when running client-side only.

Open source FingerprintJS and Fingerprint Pro are two different products. There are no IP addresses as an entropy source for open source FingerprintJS. For a full comparison check our [documentation](https://dev.fingerprint.com/docs/pro-vs-open-source).

**Reply by MentalGear** [reply]
Thank you for the detailed and quick answer !
